src/apb/engine.py

```diff
@@ -208,9 +248,25 @@ def cmdrun_init(**kwargs):
     bindable = kwargs['bindable']
     async = kwargs['async']
     params = kwargs['params']
+    skip = kwargs['skip']
 
     description = "This is a sample application generated by apb init"
 
+    # Ensure that command line options are acceptable
+    for skip_option in skip:
+        if skip_option not in SKIP_OPTIONS:
+            raise Exception(
+                "Error: [%s] is not a valid option to skip. The following are skippable options: %s" %
+                (skip_option, SKIP_OPTIONS)
+            )
```

Comment:
I would've done `--no-provision` `--no-deprovision` etc vs `--skip provision` or `--skip-provision`, `--skip-deprovision`. This avoids having to do the validation of user input. You could still do it in an automated fashion so that you can add skippable items without having to duplicate a bunch of code.

```
SKIP_OPTIONS = ['provision', 'deprovision', 'bind', 'unbind', 'roles']

for opt in SKIP_OPTIONS:
    subcmd.add_argument(
        '--skip-%s' % opt, action='store_false', dest='skip-%s' % opt,
        help=u'Specify which playbooks to not generate by default.'
    )
```
I *think* that will work. But then the user doesn't have to remember to type in the correct string and you can remove the validation altogether. It also allows you to add more to SKIP_OPTIONS in the future without having to update the generation code.